JSDoc `function` type does not accept parameters

**TypeScript Version:**  nightly (2.4.0-dev.20170513)

**Code**

```ts
// @ts-check
/** @type {function} */
const x = (a) => a + 1;
x(1);
```

**Expected behavior:**

No error.

**Actual behavior:**

`src/a.js(4,11): error TS2346: Supplied parameters do not match any signature of call target.`

It works with `Function`, but see #15825.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -6855,6 +6855,7 @@ namespace ts {
                     case "Object":
                         return anyType;
                     case "Function":
+                    case "function":
                         return globalFunctionType;
                     case "Array":
                     case "array":

--- a/src/compiler/parser.ts
+++ b/src/compiler/parser.ts
@@ -6075,7 +6075,10 @@ namespace ts {
                     case SyntaxKind.OpenBraceToken:
                         return parseJSDocRecordType();
                     case SyntaxKind.FunctionKeyword:
-                        return parseJSDocFunctionType();
+                        if (lookAhead(nextTokenIsOpenParen)) {
+                            return parseJSDocFunctionType();
+                        }
+                        break;
                     case SyntaxKind.DotDotDotToken:
                         return parseJSDocVariadicType();
                     case SyntaxKind.NewKeyword:

--- a/tests/baselines/reference/jsDocTypeTag2.js
+++ b/tests/baselines/reference/jsDocTypeTag2.js
@@ -431,28 +431,8 @@
           "kind": "space"
         },
         {
-          "text": "(",
-          "kind": "punctuation"
-        },
-        {
-          "text": ")",
-          "kind": "punctuation"
-        },
-        {
-          "text": " ",
-          "kind": "space"
-        },
-        {
-          "text": "=>",
-          "kind": "punctuation"
-        },
-        {
-          "text": " ",
-          "kind": "space"
-        },
-        {
-          "text": "any",
-          "kind": "keyword"
+          "text": "Function",
+          "kind": "localName"
         }
       ],
       "documentation": [],

--- a/tests/baselines/reference/jsDocTypes2.js
+++ b/tests/baselines/reference/jsDocTypes2.js
@@ -11,6 +11,10 @@ anyT1 = "hi";
 const x = (a) => a + 1;
 x(1);
 
+/** @type {function} */
+const y = (a) => a + 1;
+x(1);
+
 /** @type {function (number)} */
 const x1 = (a) => a + 1;
 x1(0);
@@ -28,6 +32,9 @@ var anyT1 = 2;
 anyT1 = "hi";
 /** @type {Function} */
 var x = function (a) { return a + 1; };
+x(1);
+/** @type {function} */
+var y = function (a) { return a + 1; };
 x(1);
 /** @type {function (number)} */
 var x1 = function (a) { return a + 1; };

--- a/tests/baselines/reference/jsDocTypes2.symbols
+++ b/tests/baselines/reference/jsDocTypes2.symbols
@@ -20,21 +20,30 @@ const x = (a) => a + 1;
 x(1);
 >x : Symbol(x, Decl(0.js, 9, 5))
 
+/** @type {function} */
+const y = (a) => a + 1;
+>y : Symbol(y, Decl(0.js, 13, 5))
+>a : Symbol(a, Decl(0.js, 13, 11))
+>a : Symbol(a, Decl(0.js, 13, 11))
+
+x(1);
+>x : Symbol(x, Decl(0.js, 9, 5))
+
 /** @type {function (number)} */
 const x1 = (a) => a + 1;
->x1 : Symbol(x1, Decl(0.js, 13, 5))
->a : Symbol(a, Decl(0.js, 13, 12))
->a : Symbol(a, Decl(0.js, 13, 12))
+>x1 : Symbol(x1, Decl(0.js, 17, 5))
+>a : Symbol(a, Decl(0.js, 17, 12))
+>a : Symbol(a, Decl(0.js, 17, 12))
 
 x1(0);
->x1 : Symbol(x1, Decl(0.js, 13, 5))
+>x1 : Symbol(x1, Decl(0.js, 17, 5))
 
 /** @type {function (number): number} */
 const x2 = (a) => a + 1;
->x2 : Symbol(x2, Decl(0.js, 17, 5))
->a : Symbol(a, Decl(0.js, 17, 12))
->a : Symbol(a, Decl(0.js, 17, 12))
+>x2 : Symbol(x2, Decl(0.js, 21, 5))
+>a : Symbol(a, Decl(0.js, 21, 12))
+>a : Symbol(a, Decl(0.js, 21, 12))
 
 x2(0);
->x2 : Symbol(x2, Decl(0.js, 17, 5))
+>x2 : Symbol(x2, Decl(0.js, 21, 5))
 

--- a/tests/baselines/reference/jsDocTypes2.types
+++ b/tests/baselines/reference/jsDocTypes2.types
@@ -29,6 +29,20 @@ x(1);
 >x : Function
 >1 : 1
 
+/** @type {function} */
+const y = (a) => a + 1;
+>y : Function
+>(a) => a + 1 : (a: any) => any
+>a : any
+>a + 1 : any
+>a : any
+>1 : 1
+
+x(1);
+>x(1) : any
+>x : Function
+>1 : 1
+
 /** @type {function (number)} */
 const x1 = (a) => a + 1;
 >x1 : (arg0: number) => any

--- a/tests/cases/conformance/salsa/jsDocTypes2.ts
+++ b/tests/cases/conformance/salsa/jsDocTypes2.ts
@@ -14,6 +14,10 @@ anyT1 = "hi";
 const x = (a) => a + 1;
 x(1);
 
+/** @type {function} */
+const y = (a) => a + 1;
+x(1);
+
 /** @type {function (number)} */
 const x1 = (a) => a + 1;
 x1(0);

